vaadin-notification-flow-parent/vaadin-notification-flow-integration-tests/src/test/java/com/vaadin/flow/component/notification/tests/NotificationWithClassNamesIT.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.notification.tests;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import com.vaadin.flow.component.notification.testbench.NotificationElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.tests.AbstractComponentIT;
+
+@TestPath("vaadin-notification/notification-class-names-test")
+public class NotificationWithClassNamesIT extends AbstractComponentIT {
+
+    public static final String NOTIFICATION_CARD = "vaadin-notification-card";
+
+    @Before
+    public void init() {
+        open();
+    }
+
+    @Test
+    public void openNotification_cardHasSameClassNames() {
+        findElement(By.id("open-notification-btn")).click();
+
+        waitForElementPresent(By.tagName(NOTIFICATION_CARD));
+        NotificationElement notification = $(NotificationElement.class).first();
+
+        WebElement card = $(NOTIFICATION_CARD).first();
+
+        String cardClassNames = card.getAttribute("class");
+        String notificationClassNames = notification.getAttribute("class");
+
+        Assert.assertEquals("custom", notificationClassNames);
+        Assert.assertEquals("custom", cardClassNames);
+    }
+
+    @Test
+    public void openNotification_cardChangeClassName() {
+        findElement(By.id("open-notification-btn")).click();
+
+        waitForElementPresent(By.tagName(NOTIFICATION_CARD));
+
+        findElement(By.id("add-notification-btn")).click();
+
+        WebElement card = $(NOTIFICATION_CARD).first();
+        NotificationElement notification = $(NotificationElement.class).first();
+
+        String cardClassNames = card.getAttribute("class");
+        String notificationClassNames = notification.getAttribute("class");
+
+        Assert.assertEquals("custom added", notificationClassNames);
+        Assert.assertEquals("custom added", cardClassNames);
+    }
+
+    @Test
+    public void openNotification_cardNoClassNameAfterClearClassName() {
+        findElement(By.id("open-notification-btn")).click();
+
+        waitForElementPresent(By.tagName(NOTIFICATION_CARD));
+
+        findElement(By.id("clear-notification-btn")).click();
+
+        WebElement card = $(NOTIFICATION_CARD).first();
+        NotificationElement notification = $(NotificationElement.class).first();
+
+        String cardClassNames = card.getAttribute("class");
+        String notificationClassNames = notification.getAttribute("class");
+
+        Assert.assertEquals("", notificationClassNames);
+        Assert.assertEquals("", cardClassNames);
```

Comment:
This repeating code could be refactored into a method e.g. `assertClassEquals(notification, classValue)`.
 
Also for getting notification card you could use the following code instead of querying the DOM:

```java
private TestBenchElement getCard(Notification notification) {
    return ((TestBenchElement) notification.getContext());
}
```